Library/Homebrew/test/formulary_spec.rb

```diff
@@ -0,0 +1,215 @@
+        expect(f).to be_kind_of(Formula)
+        expect(f.build).to be_kind_of(Tab)
+      end
+
+      it "returns a Formula when given a Keg" do
+        shutup do
+          installer.install
+        end
+
+        keg = Keg.new(formula.prefix)
+        f = described_class.from_keg(keg)
+        expect(f).to be_kind_of(Formula)
+        expect(f.build).to be_kind_of(Tab)
+      end
+    end
+
+    context "from Tap" do
+      let(:tap) { Tap.new("homebrew", "foo") }
+      let(:path) { tap.path/"#{name}.rb" }
+
+      it "returns a Formula when given a name" do
+        expect(described_class.factory(name)).to be_kind_of(Formula)
+      end
+
+      it "returns a Formula from an Alias path" do
+        alias_dir = tap.path/"Aliases"
+        alias_dir.mkpath
+        FileUtils.ln_s path, alias_dir/"bar"
+        expect(described_class.factory("bar")).to be_kind_of(Formula)
+      end
+
+      it "raises an error when the Formula cannot be found" do
+        expect {
+          described_class.factory("#{tap}/not_existed_formula")
+        }.to raise_error(TapFormulaUnavailableError)
+      end
+
+      it "returns a Formula when given a fully qualified name" do
+        expect(described_class.factory("#{tap}/#{name}")).to be_kind_of(Formula)
+      end
+
+      it "raises an error if a Formula is in multiple Taps" do
+        begin
+          another_tap = Tap.new("homebrew", "bar")
+          (another_tap.path/"#{name}.rb").write content
+          expect {
+            described_class.factory(name)
+          }.to raise_error(TapFormulaAmbiguityError)
+        ensure
+          another_tap.path.rmtree
+        end
+      end
+    end
+  end
+
+  specify "::from_contents" do
+    expect(described_class.from_contents(name, path, content)).to be_kind_of(Formula)
+  end
+
+  specify "::to_rack" do
+    expect(described_class.to_rack(name)).to eq(HOMEBREW_CELLAR/name)
+
+    (HOMEBREW_CELLAR/name).mkpath
+    expect(described_class.to_rack(name)).to eq(HOMEBREW_CELLAR/name)
+
+    expect {
+      described_class.to_rack("a/b/#{name}")
+    }.to raise_error(TapFormulaUnavailableError)
+  end
+
+  describe "::find_with_priority" do
+    let(:core_path) { CoreTap.new.formula_dir/"#{name}.rb" }
+    let(:tap) { Tap.new("homebrew", "foo") }
+    let(:tap_path) { tap.path/"#{name}.rb" }
+
+    before(:each) do
+      core_path.write content
+      tap_path.write content
+    end
+
+    it "prioritizes core Formulae" do
+      formula = described_class.find_with_priority(name)
+      expect(formula).to be_kind_of(Formula)
+      expect(formula.path).to eq(core_path)
+    end
+
+    it "prioritizes Formulae from pinned Taps" do
+      begin
+        tap.pin
+        formula = shutup do
+          described_class.find_with_priority(name)
+        end
+        expect(formula).to be_kind_of(Formula)
+        expect(formula.path).to eq(tap_path.realpath)
+      ensure
+        tap.pinned_symlink_path.parent.parent.rmtree
```

Comment:
Will this not automatically be removed when we delete a bunch of paths between tests?